### PR TITLE
Implement BoardFormatter with padding

### PR DESCRIPTION
**Pull request by hkgumbs**
While extending BoardFormatter to deal with 4*4, it made sense to implement with respect to dynamic board sizes and knock out padding. Rules, however, is not ready to deal with new Board sizes, so it is not a playable option. Also, I plan use a separate class to deal with command-line arguments, so padding is not "playable" yet either.

### CODE REVIEW

File: src/me/hkgumbs/tictactoe/main/java/formatter/ThreeCharacterSlot.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+public class ThreeCharacterSlot implements SlotRepresentation {
+
+    @Override
+    public String compile(Board.Mark mark, int position) {
+        if (mark != null)
+            return " " + mark.toString() + " ";
+        else if (position < 10)
+            return "(" + Integer.toHexString(position) + ")";
+        else
+            return "...";
```
Comment (robertjlooby) [inline src/me/hkgumbs/tictactoe/main/java/formatter/ThreeCharacterSlot.java]: why the `...` for positions > 9?

**Comment by robertjlooby** [comment]
- I like that the `Board` class is all board state/logic now
- I think the naming is much clearer on the predicate methods (`isEmpty` etc.)

Had one question about the display of cells > 9 but other than that :+1:

File: src/me/hkgumbs/tictactoe/main/java/formatter/ThreeCharacterSlot.java (hunk 1 above)
Comment (hkgumbs) [inline src/me/hkgumbs/tictactoe/main/java/formatter/ThreeCharacterSlot.java]: If the string lengths are different, then it becomes impossible to center the board identically on every row. My other idea was to use hexadecimal, but that only hoists the problem up to 4x4. I couldn't think of a more general solution. Ideas?

**Comment by trptcolin** [comment]
:+1: on the naming improvements; nice work.

The new java 8 feature that allows default methods to be added to on an interface is interesting. Having these concrete implementations could be a bit of a Dependency Inversion problem, but your implementations here are small and obvious enough that I can't imagine any other implementation of them, so I don't see it as an issue.

I think the `...` is just a placeholder for when you get to the 4x4 board in the next story, right? The `Integer.toHexString` is a clever trick that'll be fine for the 4x4 board, but won't hold up for larger board sizes, so I'd recommend finding something more general. Once requirements have forced a design change along one axis, it's a good idea to insulate against future changes along that same axis. That way the next similar change is super-cheap to make.

File: src/me/hkgumbs/tictactoe/main/java/formatter/ThreeCharacterSlot.java (hunk 1 above)
Comment (trptcolin) [inline src/me/hkgumbs/tictactoe/main/java/formatter/ThreeCharacterSlot.java]: Heh, we were commenting at the same time. Seems like the board size may need to determine the # of characters in the printed representation.

**Comment by trptcolin** [comment]
:+1: looks good to me.

**Comment by trptcolin** [comment]
@hkgumbs by the way, we probably didn't mention explicitly, but feel free to merge these yourself once we're all :+1: on it.

**Comment by hkgumbs** [comment]
@trptcolin ok sounds great

**Pull request merged**